ui/shared/components/panel/genes/LocusListDetail.jsx

```diff
@@ -0,0 +1,136 @@
+      rawData: locusList.intervals,
+      processRow: interval => ([interval.chrom, interval.start, interval.end, interval.genomeVersion]),
+    },
+  }]
+  const sharedFieldProps = {
+    idField: 'locusListGuid',
+    initialValues: locusList,
+    onSubmit,
+    isEditable: locusList.canEdit,
+    showEmptyValues: true,
+  }
+  return (
+    <div>
+      <Grid>
+        {FIELDS.map(({ isEditable, width, ...fieldProps }) =>
+          <CompactColumn key={fieldProps.field} width={Math.max(width, 2)}>
+            <BaseFieldView
+              {...fieldProps}
+              {...sharedFieldProps}
+              isEditable={locusList.canEdit && isEditable}
+              modalTitle={`Edit ${fieldProps.fieldName} for "${locusList.name}"`}
+            />
+          </CompactColumn>,
+        )}
+      </Grid>
+      <Header size="medium" dividing>
+        <BaseFieldView
+          {...GENE_FIELD}
+          {...sharedFieldProps}
+          modalTitle={`Edit Genes for "${locusList.name}"`}
+          compact
+          showErrorPanel
+        />
+        <ExportTableButton downloads={geneExportDownloads} buttonText="Download" float="right" fontWeight="300" fontSize=".75em" />
+      </Header>
+      <Grid columns={12} divided="vertically">
+        {(locusList.genes || []).map(gene =>
+          <Grid.Column key={gene.geneId}><ShowGeneModal gene={gene} /></Grid.Column>,
+        )}
+      </Grid>
+      <Header size="medium" dividing>
+        <BaseFieldView
+          {...INTERVAL_FIELD}
+          {...sharedFieldProps}
+          modalTitle={`Edit Intervals for "${locusList.name}"`}
+          compact
+          showErrorPanel
+        />
+        <ExportTableButton downloads={intervalExportDownloads} buttonText="Download" float="right" fontWeight="300" fontSize=".75em" />
+      </Header>
+      <Grid columns={8} divided="vertically">
+        {(locusList.intervals || []).map(interval =>
+          <Grid.Column key={interval.locusListIntervalGuid}>chr{interval.chrom}:{interval.start}-{interval.end}</Grid.Column>,
+        )}
+      </Grid>
+    </div>
+  )
+}
+
+LocusListDetail.propTypes = {
+  locusList: PropTypes.object,
+  onSubmit: PropTypes.func,
+  locusListGuid: PropTypes.string,
+  genes: PropTypes.array,
+}
+
+
+const LoadedLocusListDetail = ({ locusListGuid, locusList, projectGuid, onSubmit }) =>
+  <LocusListGeneLoader locusListGuid={locusListGuid} locusList={locusList} projectGuid={projectGuid}>
+    <LocusListDetail locusList={locusList} onSubmit={onSubmit} />
+  </LocusListGeneLoader>
+
+LoadedLocusListDetail.propTypes = {
+  locusListGuid: PropTypes.string.isRequired,
+  locusList: PropTypes.object,
+  onSubmit: PropTypes.func,
+  projectGuid: PropTypes.string,
+}
+
+const mapStateToProps = (state, ownProps) => ({
+  locusList: getLocusListsByGuid(state)[ownProps.locusListGuid] || {},
+})
+
+const mapDispatchToProps = {
+  onSubmit: updateLocusList,
+}
+
+export { LocusListDetail }
```

Comment:
Should this be export {  LoadedLocusListDetail }?

Reply:
no, thats exported as the default on the next line, this is intentionally exporting the unconnected component for testing